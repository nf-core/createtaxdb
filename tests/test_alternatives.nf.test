nextflow_pipeline {

    name "Test pipeline: NFCORE_CREATETAXDB"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_createtaxdb"
    tag "test_alternatives"
    profile "test_alternatives"

    test("-profile test_alternatives") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                        "kaiju/database.fmi - minimum file size: ${file("$outputDir/kaiju/database-kaiju.fmi").length() >= 3519604}",
                        "kaiju/database.bwt - minimum file size: ${file("$outputDir/kaiju/database-kaiju.bwt").length() >= 2612295}",
                        "kaiju/database.sa - minimum file size: ${file("$outputDir/kaiju/database-kaiju.sa").length() >= 450917}",
                        "krakenuniq/database-krakenuniq/database.idx - minimum file size: ${file("$outputDir/krakenuniq/database-krakenuniq/database.idx").length() >= 536870928}",
                        file("$outputDir/krakenuniq/database-krakenuniq/database.kdb"),
                        file("$outputDir/krakenuniq/database-krakenuniq/taxDB"),
                        file("$outputDir/krakenuniq/database-krakenuniq/database0.kdb"),
                        "krakenuniq/database-krakenuniq/database-build.log - contains string: ${path("$outputDir/krakenuniq/database-krakenuniq/database-build.log").readLines().last().contains('eval db_sort')}",
                        file("$outputDir/krakenuniq/database-krakenuniq/database.idx"),
                        file("$outputDir/krakenuniq/database-krakenuniq/database.jdb"),
                        file("$outputDir/krakenuniq/database-krakenuniq/database.kdb"),
                        file("$outputDir/krakenuniq/database-krakenuniq/database.kdb.counts"),
                        "krakenuniq/database-krakenuniq/database.kraken.tsv - contains string: ${path("$outputDir/krakenuniq/database-krakenuniq/database.kraken.tsv").readLines().any{ it.contains('NZ_LS483480.1')}}",
                        "krakenuniq/database-krakenuniq/database.report.tsv - contains string: ${path("$outputDir/krakenuniq/database-krakenuniq/database.report.tsv").readLines().last().contains('unclassified')}",
                        file("$outputDir/krakenuniq/database-krakenuniq/library/chrMT.fna"),
                        file("$outputDir/krakenuniq/database-krakenuniq/library/GCA_011545545.1_ASM1154554v1_genomic.fna"),
                        file("$outputDir/krakenuniq/database-krakenuniq/library/GCF_000292685.1_ASM29268v1_genomic.fna"),
                        file("$outputDir/krakenuniq/database-krakenuniq/library/GCF_016889925.1_ASM1688992v1_genomic.fna"),
                        file("$outputDir/krakenuniq/database-krakenuniq/library/GCF_900478275.1_34211_D02_genomic.fna"),
                        "krakenuniq/database-krakenuniq/library-files.txt - contains string: ${path("$outputDir/krakenuniq/database-krakenuniq/library-files.txt").readLines().contains('library/GCF_900478275.1_34211_D02_genomic.fna')}",
                        file("$outputDir/krakenuniq/database-krakenuniq/nucl2tax.map"),
                        file("$outputDir/krakenuniq/database-krakenuniq/seqid2taxid.map"),
                        file("$outputDir/krakenuniq/database-krakenuniq/taxDB"),
                        file("$outputDir/krakenuniq/database-krakenuniq/taxonomy/names.dmp"),
                        file("$outputDir/krakenuniq/database-krakenuniq/taxonomy/nodes.dmp"),
                        "malt/malt-build.log - contains string: ${path("$outputDir/malt/malt-build.log").readLines().any{ it.contains('--mapDB')}}",
                        "malt/malt-build.log - contains string: ${path("$outputDir/malt/malt-build.log").readLines().last().contains('Peak memory')}",
                        path("$outputDir/malt/malt_index/index0.idx"),
                        "malt/malt_index/ref.db - minimum file size: ${file("$outputDir/malt/malt_index/ref.db").length() >= 3241218}",
                        "malt/malt_index/ref.idx- minimum file size: ${file("$outputDir/malt/malt_index/ref.idx").length() >= 48}",
                        path("$outputDir/malt/malt_index/ref.inf"),
                        path("$outputDir/malt/malt_index/taxonomy.idx"),
                        path("$outputDir/malt/malt_index/taxonomy.map"),
                        path("$outputDir/malt/malt_index/taxonomy.tre"),
                        "downstream_samplesheets/databases-taxprofiler.csv - nr. lines: ${path("$outputDir/downstream_samplesheets/databases-taxprofiler.csv").readLines().size() == 5}",
                    ).match()
                },
                { assert new File("$outputDir/pipeline_info/nf_core_createtaxdb_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert path("$outputDir/malt/malt_index/table0.db").exists() },
                { assert path("$outputDir/malt/malt_index/table0.idx").exists() },
            )
        }
    }
}
