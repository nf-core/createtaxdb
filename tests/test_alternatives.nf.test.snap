{
    "test_alternatives_profile": {
        "content": [
            "database.fmi:md5,54fd89f5e4eab61af30175e8aa389598",
            "database.bwt:md5,fe639339371f5b4352017522f54d0c99",
            "database.sa:md5,7d4a6481ad5ffdc2058bb35331a64d4d",
            "database.idx",
            "database.kdb:md5,a24fce43bedbc6c420f6e36d10c112a3",
            "taxDB:md5,1aed1afa948daffc236deba1c5d635db",
            "database0.kdb:md5,a24fce43bedbc6c420f6e36d10c112a3",
            "database-build.log",
            "database.idx:md5,60b87374f00c4661fe1c9d3448eb2d97",
            "database.jdb:md5,807994585f56c0633e9d0418ae967434",
            "database.kdb:md5,a24fce43bedbc6c420f6e36d10c112a3",
            "database.kdb.counts:md5,33e5bec37e0e1cab9844ebf65642285f",
            "database.kraken.tsv:md5,364c1ee780bc0952a59d0d00734cc000",
            "database.report.tsv",
            "haemophilus_influenzae.fna:md5,f9f0fbb922adaf3968f8edbfbafe0bd8",
            "sarscov2.fasta:md5,6e9fe4042a72f2345f644f239272b7e6",
            "library-files.txt:md5,e99416d72b5fc9ff2c3a3c6fed3c8be3",
            "nucl2tax.map:md5,f8963dc677beaecd50743bb9133547da",
            "seqid2taxid.map:md5,d41d8cd98f00b204e9800998ecf8427e",
            "taxDB:md5,1aed1afa948daffc236deba1c5d635db",
            "names.dmp:md5,7636e19ba77a6b41a8be70f6dcf278e4",
            "nodes.dmp:md5,05a9d260c11234889f7378845efbd951",
            true,
            "index0.idx:md5,ef349ebc456d6446b75570b8a0e57c0d",
            "ref.db:md5,9372b49e5990fb4b29ed986a1fc737c4",
            "ref.idx:md5,b99c3dc612948a7e4e490c3c74498c80",
            "ref.inf:md5,e7ae0e08f730a851abba085e6cc7b4ec",
            "taxonomy.idx:md5,f6c520613e1154909658f976498dd4a8",
            "taxonomy.map:md5,5bb3f2192e925bca2e61e4b54f1671e0",
            "taxonomy.tre:md5,f76fb2d5aa9b0d637234d48175841e0e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-01-17T11:33:20.372559338"
    }
}